lib/twingly/url.rb

```diff
@@ -103,11 +108,26 @@ def try_addressable_normalize(addressable_uri)
         raise
       end
 
+      def valid_hostname?(domain)
+        # No need to check the TLD, the public suffix list does that
+        labels = domain.split(DOT)[..-2].map(&:to_s)
```

Comment:
I prefer this, but it's highly subjective :)

```suggestion
        labels = domain.split(DOT)[...-1].map(&:to_s)
```